tests/unit/mongo.spec.ts

```diff
@@ -3657,6 +3657,32 @@ describe('Pipeline to mongo translator', () => {
     ]);
   });
 
+  it('should fail if regexes are used in conditions', () => {
+    expect(() =>
+      mongo36translator.translate([
+        {
+          name: 'ifthenelse',
+          newColumn: 'NEW_COL',
+          if: { column: 'TEST_COL', operator: 'matches', value: '^a' },
+          then: '"True"',
+          else: '"False"',
+        },
+      ]),
+    ).toThrow('Unsupported operator');
+
+    expect(() =>
+      mongo36translator.translate([
+        {
+          name: 'ifthenelse',
+          newColumn: 'NEW_COL',
+          if: { column: 'TEST_COL', operator: 'notmatches', value: '^a' },
+          then: '"True"',
+          else: '"False"',
+        },
+      ]),
+    ).toThrow('Unsupported operator');
+  });
+
```

Comment:
Est ce que ce test ne devrait pas aussi être fait pour mongo 4.0 ?

Reply:
Indeed, I've been a bit lazy and wanted to avoid copy/pasting these tests, as Mongo40 inherits directly from Mongo36.
Di you think we should copy this test to mongo 4.0 test suite?

Reply:
I doesn't cost much but also doesn't provide a lot of value. 
The best setup would be parameterized tests for common translator behaviors but if this does not exist we'll do without.

Reply:
It's an excellent idea to parametrize tests with conditional version support. And it's also really easy ^^
I've done it in 65294bd ; what do you think?